source/Irrlicht/CSceneManager.h

```diff
@@ -261,22 +261,23 @@ namespace scene
 		struct DefaultNodeEntry
 		{
 				DefaultNodeEntry(ISceneNode* n) :
-					Node(n), renderPriority(0x80000000u), Material(video::EMT_SOLID)
+					Node(n), renderPriority(0x80000000u)//, Material(video::EMT_SOLID)
 				{
 					renderPriority = n->getRenderPriorityScore();
-					if (n->getMaterialCount())
-						Material = n->getMaterial(0).MaterialType;
+					//if (n->getMaterialCount())
+					//	Material = n->getMaterial(0).MaterialType;
 				}
 
 				bool operator < (const DefaultNodeEntry& other) const
 				{
-					return (renderPriority < other.renderPriority)||(renderPriority==other.renderPriority && Material<other.Material);
+                    //TODO ??
+					return (renderPriority < other.renderPriority)/*||(renderPriority==other.renderPriority && Material<other.Material)*/;
 				}
 
 				ISceneNode* Node;
 			private:
 				uint32_t renderPriority;
-				video::E_MATERIAL_TYPE Material;
+				//video::E_MATERIAL_TYPE Material;

```

Comment:
in the future: sort by pipeline pointer

Reply:
@Crisspl could sort by pipeline shaders, then raster parameters, then descriptor set pointers

Reply:
progress?

Reply:
sort what? whats going on? theres no pipeline ptr in there anyway. seriously WHAT

Reply:
instead of sorting on the material associated with the first meshbuffer in the node, sort by graphics pipeline of the first meshbuffer (most significant are bound shaders [geometry>tessellation>fragment>vertex], then raster params, then anything else).

Reply:
@Crisspl can you back these comments up somewhere, I'm scared I'll have to go over it again in the near future when you fix `irr::scene`